packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart

```diff
@@ -47,6 +49,21 @@ void restoreFuchsiaPortForwardingFunction() {
   fuchsiaPortForwardingFunction = _SshPortForwarder.start;
 }
 
+/// A general error raised when something fails within a
+/// [FuchsiaRemoteConnection].
+class FuchsiaRemoteConnectionError extends Error {
+  /// Basic constructor outlining the reason for the failure in `message`.
+  FuchsiaRemoteConnectionError(this.message);
+
+  /// The reason for the failure.
+  final String message;
+
+  @override
+  String toString() {
+    return '$FuchsiaRemoteConnectionError: $message\n${super.stackTrace}';
```

Comment:
The stack trace should show up if it's uncaught - you shouldn't need to add it to `toString()`.  If it's caught and being logged, then the place where it's caught should record the stack trace.

Reply:
Fixed.